src/pathfind/teleport.cpp

```diff
@@ -215,56 +215,41 @@ teleport_map::teleport_map(
 			}
 		}
 
-		std::string teleport_id = group.get_teleport_id();
 		std::set<map_location>::iterator source_it = locations.first.begin();
 		for (; source_it != locations.first.end(); ++source_it ) {
-			if(teleport_map_.count(*source_it) == 0) {
-				std::set<std::string> id_set;
-				id_set.insert(teleport_id);
-				teleport_map_.emplace(*source_it, id_set);
+			auto map_it = teleport_map_.find(*source_it);
+
+			if(map_it == teleport_map_.end()) {
+				std::unordered_set<map_location> map_locations;
+				map_locations.insert(locations.second.begin(), locations.second.end());
+				teleport_map_.emplace(*source_it, map_locations);
 			} else {
-				(teleport_map_.find(*source_it)->second).insert(teleport_id);
+				map_it->second.insert(locations.second.begin(), locations.second.end());
 			}
 		}
-		sources_.emplace(teleport_id, locations.first);
-		targets_.emplace(teleport_id, locations.second);
+		sources_.insert(locations.first.begin(), locations.first.end());
+		targets_.insert(locations.second.begin(), locations.second.end());
 	}
 }
 
-std::set<map_location> teleport_map::get_adjacents(map_location loc) const
+const std::unordered_set<map_location>& teleport_map::get_adjacents(map_location loc) const
```

Comment:
Should take a reference.

Reply:
With `map_location` I don't think it matters. It's just a pair of integers.

Reply:
I also dont see why it would be better if it was a reference

Reply:
Alright, it's fine to leave.